src/PresentationalComponents/ExecutiveReport/Download.js

```diff
@@ -0,0 +1,36 @@
+import './_Download.scss';
+
+import { RULES_FETCH_URL, STATS_REPORTS_FETCH_URL, STATS_SYSTEMS_FETCH_URL } from '../../AppConstants';
+
+import API from '../../Utilities/Api';
+import { DownloadButton } from '@redhat-cloud-services/frontend-components-pdf-generator';
+import ExportIcon from '@patternfly/react-icons/dist/js/icons/export-icon';
+import React from 'react';
+import buildExecReport from './Build';
+import messages from '../../Messages';
+import { useIntl } from 'react-intl';
+
+const DownloadExecReport = () => {
+    const intl = useIntl();
+    const [loading, setLoading] = React.useState(false);
+
+    const dataFetch = async () => {
+        setLoading(true);
+        const statsSystems = (await API.get(STATS_SYSTEMS_FETCH_URL)).data;
+        const statsReports = (await API.get(STATS_REPORTS_FETCH_URL)).data;
+        const topActiveRec = (await API.get(RULES_FETCH_URL, {}, { limit: 3, sort: '-impacted_count' })).data;
```

Comment:
Since these endpoints are not depending on each other, you can improve the speed of download by using Promise.all
```JS
const [statsSystems, statsReports, topActiveRec] = await Promise.all([
  (await API.get(STATS_SYSTEMS_FETCH_URL)).data,
  (await API.get(STATS_REPORTS_FETCH_URL)).data,
  (await API.get(RULES_FETCH_URL, {}, { limit: 3, sort: '-impacted_count' })).data
]);
```

Reply:
GOOD CATCH!! yah know ive always wanted to use promise.all 😀